frontend/public/components/modals/configure-update-strategy-modal.tsx

```diff
@@ -7,9 +7,12 @@ import { DeploymentModel } from '../../models';
 import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory/modal';
 import { pluralize, withHandlePromise, HandlePromiseProps } from '../utils';
 import { RadioInput } from '../radio';
+import { useTranslation } from 'react-i18next';
+import i18next from 'i18next';
 
-export const UPDATE_STRATEGY_DESCRIPTION =
-  'How should the pods be replaced when a new revision is created?';
+export const UPDATE_STRATEGY_DESCRIPTION = i18next.t(
```

Comment:
You can't use `i18next.t` to set a top-level constant outside a function or component since it will never be updated if the language changes. It is probably better to either make the description a React component or inline it where it's used.

Reply:
`UPDATE_STRATEGY_DESCRIPTION` is also used in  https://github.com/openshift/console/blob/master/frontend/packages/console-shared/src/components/dynamic-form/fields.tsx#L170
and function useSchemaDescription requires the description to be string

Reply:
> Hi @spadgett
> `UPDATE_STRATEGY_DESCRIPTION` is also used in https://github.com/openshift/console/blob/master/frontend/packages/console-shared/src/components/dynamic-form/fields.tsx#L170
> and function `useSchemaDescription` requires the description param to be string
> any suggestion to resolve this?

I'd suggest making it a React component instead of a constant string.

```ts
const UpdateStrategyDescription: React.FC = () => {
  const { t } = useTranslation();
  return t('modal~How should the pods be replaced when a new revision is created?');
};
```

Reply:
Hi @spadgett 
I tried following
```
const UpdateStrategyDescription: React.FC = () => {
  const { t } = useTranslation();
  return t('modal~How should the pods be replaced when a new revision is created?');
};
```

Reply:
No, export the `UpdateStrategyDescription` component and always use that instead.